File: week07/mapreduce/hadoop/wordcount_scala/src/main/scala/wordcount/WordCount.scala
package wordcount

import org.apache.hadoop.conf.Configured
import org.apache.hadoop.fs.Path
import org.apache.hadoop.io.{IntWritable, Text}
import org.apache.hadoop.mapreduce.Job
import org.apache.hadoop.mapreduce.lib.input.FileInputFormat
import org.apache.hadoop.mapreduce.lib.output.FileOutputFormat
import org.apache.hadoop.util.Tool

class WordCount extends Configured with Tool {

  override def run(args: Array[String]): Unit = {
    val job = Job.getInstance(getConf)
    job.setJarByClass(this.getClass)

    job.setMapperClass(classOf[WordCountMapper])
    job.setReducerClass(classOf[WordCountReducer])
    job.setCombinerClass(classOf[WordCountReducer])

    job.setMapOutputKeyClass(classOf[Text])
    job.setMapOutputValueClass(classOf[IntWritable])

    FileInputFormat.addInputPath(job, new Path(args[0]))
    FileOutputFormat.setOutputPath(job, new Path(args[0] + ".out"))

    job.waitForCompletion(true)
  } // end method run.

} // end class WordCount

File: week07/mapreduce/hadoop/wordcount_java/src/main/java/wordcount/WordCount.java
package wordcount;

import org.apache.hadoop.conf.Configured;
import org.apache.hadoop.fs.Path;
import org.apache.hadoop.io.IntWritable;
import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapreduce.Job;
import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
import org.apache.hadoop.mapreduce.lib.output.FileOutputFormat;
import org.apache.hadoop.util.Tool;
import org.apache.hadoop.util.ToolRunner;

public class WordCount extends Configured implements Tool {

  public static void main(String[] args) throws Exception {
    ToolRunner.run(new WordCount(), args);
  }

  public int run(String[] strings) throws Exception {
    Job job = Job.getInstance(getConf());
    job.setJarByClass(WordCount.class);

    job.setMapperClass(WordCountMapper.class);
    job.setReducerClass(WordCountReducer.class);
    job.setCombinerClass(WordCountReducer.class);

    job.setMapOutputKeyClass(Text.class);
    job.setMapOutputValueClass(IntWritable.class);

    FileInputFormat.addInputPath(job, new Path(strings[0]));
    FileOutputFormat.setOutputPath(job, new Path(strings[0] + ".out"));

    job.waitForCompletion(true);

    return 0;
  } // end method run.

} // end class WordCount.
